npm/src/commands/install.js

```diff
@@ -1,76 +1,90 @@
+
+import type {LibDefWithFlow} from '../lib/libDef';
 import {child_process, path, https, fs} from '../lib/node'
 
-export const name = "install";
-export const description =
-  "Installs a libdef";
+export const name = 'install';
+export const description = 'Installs a libdef to the ./flow-typed directory';
+
+export function setup(yargs: Object): Object {
+  return yargs
+    .usage(`$0 ${name} - ${description}`)
+    .options({
+      flowVersion: {
+        demand: true,
+        describe: 'The version of Flow fetched libdefs must be compatible with',
+        type: 'string',
+      },
+    });
+};
+
+function failWithMessage(message: ?string) {
+  if (message != null) {
+    console.error(message);
+  }
+  return 1;
+}
 
 type Args = {
   _: Array<string>;
   flowVersion?: string;
 }
-export async function run(args: Args): Promise<number> {
-
-  function errorNeedDefName() {
-    console.error('Please provide a libdef name ' +
-      '(example: lodash, or lodash@4.2.1)')
-  }
 
-  function errorNeedFlowVersion() {
-    console.error('Please provide a flow version ' +
-    '(example: --flow v0.23.0)')
-  }
+export async function run(args: Args): Promise<number> {
+  const { _ } = args;
 
-  if (!args._ || !(args._.length > 1)) {
-    errorNeedDefName()
-    return 1;
+  if (_ == null || !(_.length > 1)) {
+    return failWithMessage('Please provide a libdef name (example: lodash, or lodash@4.2.1)');
   }
 
-  if (!args.flow) {
-    errorNeedFlowVersion()
-    return 1;
+  if (args.flowVersion == null) {
+    return failWithMessage('Please provide a flow version (example: --flow 0.23)');
   }
 
   const term = args._[1];
 
-  const flowVersion = args.flow
+  const flowVersion = args.flowVersion;
   const matches = term.match(/(.*)@?(.*)?/)
   const defName = (matches && matches[1])
   const defVersion = (matches && matches[2]) || 'auto'
 
+  const failNoLibDefFound = () => failWithMessage('Sorry, no libdef found with that name, version and flow version');
   if (!defName) {
-    errorNeedDefName()
-    return 1;
+    return failNoLibDefFound(); 
   }
 
   const defs = await getGHLibsAndFlowVersions();
   const filtered = filterDefs(term, defs, flowVersion, defVersion)
 
   if (!filtered.length) {
-    console.error('Sorry, no libdef found with that name, version, ' +
-    'and flow version.')
-    return 1;
+    return failNoLibDefFound();
   }
 
   const def = filtered[0]
 
-  console.log('DEBUG: About to install', def.pkgName, def.pkgVersionStr)
+  await new Promise((resolve, reject) => {
+    const flowTypedDir = path.resolve('./flow-typed')
+
+    console.log(`File will be downloaded to '${flowTypedDir}'`);
+    mkdirp.sync(flowTypedDir)
 
-  await new Promise((res, rej) => {
-    const flowTypedFolder = path.resolve('./flow-typed')
+    const url = `https://raw.githubusercontent.com/flowtype/flow-typed/master/definitions/${def.pkgName}_${def.pkgVersionStr}/flow_${def.flowVersionStr}/${def.pkgName}_${def.pkgVersionStr}.js`;
```

Comment:
I realized that actually the fetching of the libDefs was making problems and I never reached the point where I queried `raw.githubusercontent.com` ... I added the GH-client / secret environment variable and used my own application authorization data for now.